docs/xks/developer-guide/observability.md

```diff
@@ -0,0 +1,244 @@
+
+```podlogs.yaml
+apiVersion: monitoring.grafana.com/v1alpha1
+kind: PodLogs
+metadata:
+  name: tenant-namespace-log
+  labels:
+    xkf.xenit.io/monitoring: tenant
+spec:
+  selector: {}
+  pipelineStages:
+    - cri: {}
+```
+
+### Tracing
+
+The tracing setup is a bit different compared to logging and metrics, instead of having some yaml file where you define the grafana agent how to gather
+data from it your application, you instead push data to the grafana agent on a specific URL within your application.
+
+Opentelemetry supports both http and grpc communication to gather traces from your application.
+
+To send http data we use `4318` and for gRPC we use `4317`.
+
+Assuming that you are using XKF the URL will be
+
+- [http://grafana-agent-traces.opentelemetry.svc.cluster.local:4318/v1/traces](http://grafana-agent-traces.opentelemetry.svc.cluster.local:4318/v1/traces)
+- [http://grafana-agent-traces.opentelemetry.svc.cluster.local:4317/v1/traces](http://grafana-agent-traces.opentelemetry.svc.cluster.local:4317/v1/traces)
+
+#### Tail-based sampling
+
+By default the grafana-agent that is deployed by XKF forwards all traces without any special config.
+This can cause high costs thanks to the amount of data that is sent. Grafana-agent have done there own solution on `probabilistic sampling`
+that is called [tail-based sampling](https://grafana.com/docs/tempo/latest/grafana-agent/tail-based-sampling/) that can help you solving this issue.
+
+For a easy way of managing access token you can setup your own trace agent with the custom config that you want and then have it forward all the traffic to our central trace agent in the opentelemetry namespace.
+We plan to write a blog post about this in the future but below you can find a simple example configmap that you can use together with your trace agent to send data to the central agent.
+
+```trace-agent-configmap.yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: grafana-agent-traces
+data:
+  agent.yaml: |
+    tempo:
+      configs:
+        - name: default
+          remote_write:
+            - endpoint: "grafana-agent-traces.grafana-agent.svc.cluster.local:4317"
+              insecure: true
+          receivers:
+            otlp:
+              protocols:
+                http: {}
+                grpc: {}
+            jaeger:
+              protocols:
+                grpc:
+                  endpoint: 0.0.0.0:55678
+          tail_sampling:
+            # policies define the rules by which traces will be sampled. Multiple policies
+            # can be added to the same pipeline.
+            # For more information: https://grafana.com/docs/agent/latest/configuration/traces-config/
+            # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/b2327211df976e0a57ef0425493448988772a16b/processor/tailsamplingprocessor
+            policies:
+              - probabilistic: {sampling_percentage: 90}
```

Comment:
This sounds like we are saving 10% which sound pointless. If this means what it sounds like, I think 10% would be more realistic example.